File: ios/DailyMemory/Domain/UseCase/Memory/GetMemoryUseCase.swift
import Foundation

/// Use case for getting a single memory by ID.
final class GetMemoryUseCase {
    private let memoryRepository: MemoryRepository

    init(memoryRepository: MemoryRepository = MemoryRepositoryImpl()) {
        self.memoryRepository = memoryRepository
    }

    func execute(id: String) async throws -> Memory? {
        return try await memoryRepository.getById(id)
    }
}

File: android/app/src/main/java/com/dailymemory/domain/usecase/memory/GetMemoryUseCase.kt
package com.dailymemory.domain.usecase.memory

import com.dailymemory.domain.model.Memory
import com.dailymemory.domain.repository.MemoryRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

/**
 * Use case for getting a single memory by ID.
 */
class GetMemoryUseCase @Inject constructor(
    private val memoryRepository: MemoryRepository
) {
    suspend operator fun invoke(id: String): Memory? {
        return memoryRepository.getById(id)
    }

    fun observe(id: String): Flow<Memory?> {
        return memoryRepository.observeById(id)
    }
}
